Makefile

```diff
@@ -28,7 +28,7 @@ ifdef BAZEL_REMOTE_CACHE
   BAZEL_BUILD_OPTS += --remote_cache=$(BAZEL_REMOTE_CACHE)
 endif
 
-BAZEL_TEST_OPTS ?= --jobs=HOST_RAM*.0003 --test_timeout=300
+BAZEL_TEST_OPTS ?= --jobs=HOST_RAM*.0003 --test_timeout=300 --local_test_jobs=1
```

Comment:
Unfortunately this will also limit the builds to one job, which would be a huge waste on the big runners. We'll need to split the test runs to two steps everywhere:
1. `bazel build --build_tests_only` to build the test binaries, like in the `envoy-test-deps` make target
2. Run the tests with `bazel test --jobs=1`

On the makefiles we can make the test run target depend on the test-deps target, but we'll have to make sure they are paired properly (so that the deps are built with all the same build options as the test run).

If we are seeing this same issue in the CI, then we can add the `--jobs=1` to the test run step, assuming we have build all the test code in a separate build step.